Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -152,4 +162,31 @@ private function getFilterValueFromUrl($value)
 
         return $value;
     }
+
+    /**
+     * Gets query from strategy.
+     *
+     * @param  string $strategy
+     * @param  string $property
+     * @param  string $value 
+     *
+     * @return string
+     */
+    private function getQueryFromStrategy($strategy) {
+        switch ($strategy) {
+            case self::STRATEGY_EXACT:
+                return 'o.%s = \'%s\'';
+            
+            case self::STRATEGY_PARTIAL:
+                return 'o.%s LIKE \'%%%s%%\'';
+
+            case self::STRATEGY_STARTS_WITH:
+                return 'o.%s LIKE \'%s%%\'';
+
+            case self::STRATEGY_ENDS_WITH:
+                return 'o.%s LIKE \'%%%s\'';
+        }
+
+        throw new \InvalidArgumentException(sprintf('The strategy %s does not exist.', $strategy));
```

Comment:
Moreover, maybe quoting the value (ie `The startegy "%s" does not exists.`) as it is often useful for users to see directly that the value is empty in case of error.